spdx/parsers/tagvaluebuilders.py

```diff
@@ -1435,47 +1514,47 @@ def set_snippet_comment(self, doc, comment):
         Raise CardinalityError if comment already set.
         """
         self.assert_snippet_exists()
-        if not self.snippet_comment_set:
-            self.snippet_comment_set = True
-            if validations.validate_snip_comment(comment):
-                doc.snippet[-1].comment = str_from_text(comment)
-                return True
-            else:
-                raise SPDXValueError("Snippet::SnippetComment")
-        else:
+        if self.snippet_comment_set:
             raise CardinalityError("Snippet::SnippetComment")
 
+        if not validations.validate_snip_comment(comment):
+            raise SPDXValueError("Snippet::SnippetComment")
+
+        self.snippet_comment_set = True
+        doc.snippet[-1].comment = str_from_text(comment)
+        return True
+
     def set_snippet_attribution_text(self, doc, text):
         """
         Set the snippet's attribution text .
         Raise SPDXValueError if text is not free form text.
         """
         self.assert_snippet_exists()
-        if validations.validate_snippet_attribution_text(text):
-            doc.snippet[-1].attribution_text = str_from_text(text)
-            return True
-        else:
+        if not validations.validate_snippet_attribution_text(text):
             raise SPDXValueError("Snippet::AttributionText")
 
+        doc.snippet[-1].attribution_text = str_from_text(text)
+        return True
+
     def set_snippet_copyright(self, doc, text):
         """Set the snippet's copyright text.
         Raise OrderError if no snippet previously defined.
         Raise CardinalityError if already set.
         Raise SPDXValueError if text is not one of [None, NOASSERT, TEXT].
         """
         self.assert_snippet_exists()
-        if not self.snippet_copyright_set:
-            self.snippet_copyright_set = True
-            if validations.validate_snippet_copyright(text, optional=True):
-                if isinstance(text, str):
-                    doc.snippet[-1].copyright = str_from_text(text)
-                else:
-                    doc.snippet[-1].copyright = text  # None or NoAssert
-            else:
-                raise SPDXValueError("Snippet::SnippetCopyrightText")
-        else:
+        if self.snippet_copyright_set:
             raise CardinalityError("Snippet::SnippetCopyrightText")
 
+        if not validations.validate_snippet_copyright(text, optional=True):
+            raise SPDXValueError("Snippet::SnippetCopyrightText")
+
+        self.snippet_copyright_set = True
+        if isinstance(text, str):
+            doc.snippet[-1].copyright = str_from_text(text)
+        else:
+            doc.snippet[-1].copyright = text  # None or NoAssert
```

Comment:
nit: No `return True` on this method

Reply:
Yep, and neither are there in several other of these methods. I didn't bother adding them as their merit seems questionable and everything works fine without them